### ISSUE TITLE
Sampling details for FID evaluation

### ISSUE THREAD

**Issue by zhihongp** [openai/guided-diffusion#56]
While it's mentioned that the full training set was used as reference for FID, I couldn't find details about evaluation so I wonder if I have missed them.  Noticed the 10K samples (instead of 50K) used in ablation, but no other specifications about sampling.  Besides, for guided sampling, specifics like samples per class would be useful too.  Any insight is appreciated, either from author or readers.

**Comment by zhihongp** [comment]
Found in evaluation code that it's 50K but #samples/class is random

**Comment by xcharxlie** [comment]
> Found in evaluation code that it's 50K but #samples/class is random

@zhihongp I also have the same question. Do you know why they use [random int](https://github.com/openai/guided-diffusion/blob/main/scripts/classifier_sample.py#L72) to represent the output class they desire to generate in classifier-guided sampling? I believe this "y" is the class you want to see in the outputs, right?
`classes = th.randint(
            low=0, high=NUM_CLASSES, size=(args.batch_size,), device=dist_util.dev()
        )
        model_kwargs["y"] = classes`

**Comment by wwj723452017** [comment]
> > Found in evaluation code that it's 50K but #samples/class is random
> 
> @zhihongp I also have the same question. Do you know why they use [random int](https://github.com/openai/guided-diffusion/blob/main/scripts/classifier_sample.py#L72) to represent the output class they desire to generate in classifier-guided sampling? I believe this "y" is the class you want to see in the outputs, right? `classes = th.randint( low=0, high=NUM_CLASSES, size=(args.batch_size,), device=dist_util.dev() ) model_kwargs["y"] = classes`



@zhihongp @xcharxlie 
hi,have you solved this question?I tried to look into image_sample.py script,and in muti-class situation,I want to know how the pics matching with the lables. According to (model_kwargs["y"] = classes) , classes are something like these:682 380 174 125 533  27 133 421 581 985(tensor),AND I can't figure out the numbers and the relationship with the labels.Thanks for your reply.

**Comment by sunset-clouds** [comment]
The rFID on Imagenet I got from the test is 2.70, but many people told me that the rFID of VAR is 0.92. I don’t know which step in the test went wrong.
